cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -240,24 +241,24 @@ public String commitAndPush(final CommitMeta commitMeta,
           "No changes have been made for the applications to push.");
     }
 
-    git.add().addFilepattern(fileChanged.toString()).call();
+    for (Path fileChanged : filesChanged) {
+      git.add().addFilepattern(fileChanged.toString()).call();
```

Comment:
Is it possible to get the file hash from whatever this returns? I see there is a DirCache.getEntry().getObjectId(), but not sure if that's the same as what's happening in getFileHashes.

If so it would be better than doing the tree walk later, and would allow passing in an iterator of paths instead of the Set.

Reply:
Checked in test and looks like the hashes are correct. We can skip the tree walk

Reply:
Updated the `commitandpush` take a `BiFunction<T, String, O>` and return `Collection<O>`. Updated all the test accordingly